test: E2E test setup script creates iOS simulators

dcalhoun opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->
Expand the E2E test setup script to ensure the required iOS simulators are
available.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
Relates to https://github.com/WordPress/gutenberg/issues/55251. Simplify setting
up the E2E test environment by automating the process for creating the required
iOS simulators.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
- Convert spaces to tabs to match project
- E2E test script detects or creates iOS simulators
- E2E setup script throws error if required iOS runtime unavailable
- Update E2E test environment setup documentation

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->
**With required simulators unavailable**
1. Delete the iPhone 14 and iPad (10th generation) simulators via `xcrun simctl delete <device-id>`. 
2. Run `npm run native test:e2e:setup`
3. Verify the correct iOS simulators are created. 

**With required simulators available**
1. Verity the iPhone 14 and iPad (10th generation) simulators are available via `xcrun simctl list devices`. 
2. Run `npm run native test:e2e:setup`
3. Verify duplicative simulators are not created. 

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->
n/a

## Screenshots or screencast <!-- if applicable -->
n/a

dcalhoun committed:
refactor: Convert spaces to tabs to match project

Tabs are the project configuration's preferred format.

dcalhoun committed:
test: E2E test script detects or create iOS simulators

Simplify E2E test environment setup by creating the required iOS
simulators if they are not present.

dcalhoun committed:
test: E2E setup script throws error if required iOS runtime unavailable

Provide helpful directions to fix E2E test environment setup.

dcalhoun committed:
docs: Update E2E test environment setup documentation

Adjust directions to account for the expanded capabilities of the
`test:e2e:setup` script.

dcalhoun committed:
refactor: Abbreviate E2E test setup script logs

An attempt to improve comprehension through brevity.

dcalhoun committed:
refactor: Avoid repetition

dcalhoun committed:
feat: Colorize E2E test script log

Improve notability of errors.

dcalhoun commented on packages/react-native-editor/bin/test-e2e-setup.sh:
I consider myself a shell script novice. Please provide any feedback you may have. 🙇🏻

geriux reviewed (commented):
Wow this is awesome 👏 Loving the different outputs and its levels (info/error/success) ⭐ 

- ✅  I tested creating the devices and it worked as expected
- ✅  I tested running the script with the devices already created and it worked as expected.
- ⚠️ Tried to run the script after updating the `platformVersion` to a different runtime that I don't have installed and it didn't return the [error in this check](https://github.com/WordPress/gutenberg/pull/55716/files#diff-5fb1a9937f08b0ecd6a7a847922dfb8ff40e5965762a1e59f98ddeaffb99330bR44). It continued to try to create the simulator.

I ran separately:
```
xcrun simctl list runtimes -j | jq -r --arg version "16.9" '.runtimes | to_entries[] | select(.value.version | contains($version))'
```

And it returns an empty string so maybe that's why it skipped the `if` check?

@dcalhoun if you run that command what does it return? Maybe we need to also check that the command's output is not empty 🤔

dcalhoun committed:
fix: Throw error when iOS runtime check returns empty response

The logic used to search for the runtime results in an empty response if
the runtime is not installed. Therefore, we need to verify the response
it not empty.

dcalhoun commented:
> * ⚠️ Tried to run the script after updating the `platformVersion` to a different runtime that I don't have installed and it didn't return the [error in this check](https://github.com/WordPress/gutenberg/pull/55716/files#diff-5fb1a9937f08b0ecd6a7a847922dfb8ff40e5965762a1e59f98ddeaffb99330bR44). It continued to try to create the simulator.

Good catch! Thank you for testing so thoroughly. I pushed a change that I believe resolves this issue. Will you please review?

geriux reviewed (approved):
This is a major improvement 🤩 LGTM! 🚀
